### DISCUSSION TITLE
vcpkg hangs on downloading "https://7-zip.org/a/7z1900.msi"

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vcpkg#28296] [Q&A]
Windows 10, I followed the installation instructions this afternoon to build vcpkg. `--version` gives 2022-11-10-5fdee72bc1fceca198fb1ab7589837206a8b81ba.

Trying to follow the build instructions for Minetest ([https://github.com/minetest/minetest](https://github.com/minetest/minetest)) and in the instructions, it states a command for installing the necessary dependencies using vcpkg, which it says is "highly recommended" that it be used as package manager. The command:
`vcpkg install zlib zstd curl[winssl] openal-soft libvorbis libogg libjpeg-turbo sqlite3 freetype luajit gmp jsoncpp opengl-registry --triplet x64-windows
`
It goes well until a certain point from where the results are like clockwork.
`Installing 11/20 gmp:x64-windows...
Building gmp[core]:x64-windows...
-- Downloading https://gmplib.org/download/gmp/gmp-6.2.1.tar.xz -> gmp-6.2.1.tar.xz...
-- Downloading https://gmplib.org/repo/gmp/raw-rev/5f32dbc41afc -> gmp-arm64-asm-fix-5f32dbc41afc.patch...
-- Extracting source C:/Program Files/vcpkg/downloads/gmp-6.2.1.tar.xz
-- Applying patch asmflags.patch
-- Applying patch cross-tools.patch
-- Applying patch subdirs.patch
-- Applying patch msvc_symbol.patch
-- Applying patch arm64-coff.patch
-- Applying patch C:/Program Files/vcpkg/downloads/gmp-arm64-asm-fix-5f32dbc41afc.patch
-- Using source at C:/Program Files/vcpkg/buildtrees/gmp/src/v6.2.1-0c723d4b6f.clean
-- Getting CMake variables for x64-windows
-- Downloading https://github.com/llvm/llvm-project/releases/download/llvmorg-12.0.0/LLVM-12.0.0-win64.exe -> LLVM-12.0.0-win64.7z.exe...
-- Downloading https://7-zip.org/a/7z1900.msi -> 7z1900.msi...`
And on that last line it appears to hang. I left to go do something else, wound up not returning for over an hour and it was still going on that same line. Using resource monitor, it shows that it does use network for about 30sec before the activity becomes more sporadic before it stops altogether and disappears from processes using net. It then just seems to hang there; no noticible CPU usage, no RAM, no disc; it just seems to sleep from my eyes.

Running the command again, it goes quickly back to that, recognizing the other deps as installed and then begins the same block again but it "uses cached 7z1900.msi" and at that point, it hangs again and once again I left it for over an hour and the cursor remained at the beginning of an empty line under that statement.

The downloads folder contains one "7z1900.msi" but I'm a bit worried about trying to install using it for fear it might be corrupted in some way. The docs on how to install were very vague and the part that you would only be building and have to place the whole directory where you wanted it permanently and put it on path kind of had me confused so it could be I had it wrong but I think you can see what I did from the stdout above. Opening the url in my browser does download a file of the same name and very quickly and without issue. Very strange issue I feel. I can't find any documenting of it or anything like it. Should I bypass by getting and using the desired installer manually? Will vcpkg recognize it as installed then and move on or does it do special configuration?

**Comment by JackBoosY** [comment]
You can just download `https://7-zip.org/a/7z1900.msi` manually and put it into _VCPKG_ROOT/downloads_ to avoid this.

**Reply by ghost** [reply]
Thank you for your suggestion but it only results in exactly what happens when I re-try the command and it acts as though it's trying to get the cache. Apon trying to process that file, it immediately hangs and both cpu and disk usage drop to 0 indefinitely although cmake retains some memory without doing anything

**Reply by JackBoosY** [reply]
I don't know, maybe related to your network?

**Comment by ghost** [comment]
The rest of the packages installed just fine. I was having some network issues throughout yesterday but it was fine up until then. I'll check again.

**Comment by xrr-233** [comment]
Did you solve the problem? I encountered the same problem again when installing COLMAP

**Comment by ghost** [comment]
Ok I should have done this a while ago. I gave up. I can't remember the results of canoodling with the networking. May have worked may not have. Closing for now because I didn't have the answer and not present enough to keep working the problem atm.

**Reply by xrr-233** [reply]
yeah, actually there is another issue https://github.com/microsoft/vcpkg/issues/28912 but they also cannot reproduce the bug.

**Comment by dg0yt** [comment]
FTR the download of `7z1900.msi` was replaced about one year ago, #23424.

**Reply by dg0yt** [reply]
... but not in `scripts/cmake/vcpkg_find_acquire_program.cmake` :facepalm:

**Comment by dg0yt** [comment]
For `gmp` in particular:

> -- Using cached LLVM-15.0.6-win64.7z.exe.
> -- Using cached 7z1900.msi.

I wouldn't be surprised if *extracting* `LLVM-15.0.6-win64.7z.exe` takes very long.
But this is not really what is meant: It should use clang-cl which comes with modern versions of visual studio.

**Reply by DarkSun812** [reply]
How solved this problem installing gmp??

**Reply by dg0yt** [reply]
Install the clang-cl offered by visual studio to avoid download/extraction.

**Reply by DarkSun812** [reply]
The issue is that I want to install CGAL and one of the dependencies it downloads is gmp in vcpkg, but when I take it to that library it gets stuck, I left it for about 2 hours and nothing finished. I tried to install it with the source but I could not get it to recognize the libraries in visual studio, I think I need to compile it but in windows it has been very difficult to install it, can you help me with that?

**Reply by dg0yt** [reply]
So did you "Install the clang-cl offered by visual studio to avoid download/extraction" ?

**Reply by DarkSun812** [reply]
I installed clang-cl but it told me to uninstall it to install the other gmp (using vcpkg).

**Reply by dg0yt** [reply]
clang-cl told you to uninstall gmp? I don't think the VS component will do that.

**Comment by xrr-233** [comment]
Yes you are right! After installing clang-cl in vs2019 gmp was successfully installed finally. Thank you very much!!!
